docs/website/docs/general-usage/incremental-loading.md

```diff
@@ -300,6 +300,23 @@ pipeline.run(dim_customer())  # third run — 2024-04-10 06:45:22.847403
 | 2024-04-09 18:27:53.734235 | **2024-04-10 06:45:22.847403** | 2 | bar | 2 |
 | 2024-04-09 22:13:07.943703 | NULL | 1 | foo_updated | 1 |
 
+#### Example: incremental `scd2`
+`retire_absent_rows` can be set to `False` to work with incremental extracts instead of full extracts:
+```py
+@dlt.resource(
+    merge_key="my_natural_key",
+    write_disposition={
+        "disposition": "merge",
+        "strategy": "scd2",
+        "retire_absent_rows": False,
+    }
+)
+def dim_customer():
+    ...
+...
+```
+Using this setting, records are not retired in the destination if their corresponding natural keys are not present in the source extract. This allows for incremental extracts that only contain updated records. You need to specify the natural key as `merge_key` when `retire_absent_rows` is `False`. Compound natural keys are allowed and can be specified by providing a list of column names as `merge_key`.
```

Comment:
@rudolfix we are using only the merge_key as opposed to a combination of primary key and merge key here, let us know if this is ok.

Reply:
@jorritsandbrink @sh-rp I think there's some kind of conceptual confusion. `merge_key` is correct but it is not a natural key for the source data. it is the same `merge_key` we have in `delete-insert` strategy. 

Example 1: I load data day by day. I set the merge key to `day` column. I load the same day twice. Then I retire records only from the given day.
Example 2: I set the merge key to the same value as primary (natural) key (or to the content hash if no natural key is present). Then I have the behavior as I'd set the `retire_absent_rows` to False. (which is not needed anymore btw.)
Example 3: I update chunked documents and use merge key to be doc_id. Then I retire all the missing chunks for those doc ids, not touching the others.

My take:
1. drop the flag
2. if the merge key is present, always limit the set of records to retire
3. update the documentation to explain two basic cases (do not retire deleted records at all, retire deleted records only for given partition)

WDYT? IMO this is really powerful functionality if done this way

Reply:
I think you're right :) I'll adapt per your suggestions.